frontend/src/charts/trendsChart/TrendsTooltip.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Tooltip for the charts that track trends over time
+ * @param {object[]} data array of timeseries data objects
+ * @param {string} selectedDate the date that is currently hovered
+ * @param {object} axisConfig an object containing the configuration for axes - type and labels
+ * @param {boolean} isMobile a flag to determine whether user is viewing app below the mobile breakpoint
+ * returns jsx of a div with a grid of names, bar chart viz, and amounts
+
+
+/* External Imports */
+import React, { Fragment } from "react";
+
+/* Local Imports */
+
+/* Styles */
+import styles from "./Trends.module.scss";
+
+/* Components */
+
+/* Constants */
+import { TrendsData, GroupData, GroupValues, AxisConfig } from "./types";
+import { TYPES, FORMATTERS as F, COLORS as C } from "./constants";
+
+/* Helpers */
+import {
+  getAmountsByDate,
+  sortDataDescending,
+  translateXPctShare,
+  getWidthPctShare,
+  getWidthHundredK,
+} from "./helpers";
+
+/* Define type interface */
+export interface TrendsTooltipProps {
+  data: TrendsData;
+  selectedDate: string | null;
+  axisConfig: AxisConfig;
+  isMobile: boolean;
+}
+
+/* Render component */
+export function TrendsTooltip({
+  data,
+  selectedDate,
+  axisConfig,
+  isMobile,
+}: TrendsTooltipProps) {
+  const { type, yAxisLabel = "" } = axisConfig || {};
+
+  // temp
+  const codeDictionary = {
```

Comment:
sounds like this will be replaced, but if not it could be good to move to constants

Reply:
yes will be replaced!